src_c/event.c

```diff
@@ -1586,53 +1593,36 @@ get_grab(PyObject *self)
     return PyBool_FromLong(mode);
 }
 
+/* This is a windowevent translate filter that must run before calling
+ * functions like event.clear, event.get and event.peek that deal with event
+ * type peeking/flushing in the entire event queue. This is not needed while
+ * calling functions like event.poll and event.wait because they only return
+ * the first event in the event queue, and that event can be special-cased in
+ * 'pgEvent_New' */
+static int
+_pg_translate_windowevent(void *_, SDL_Event *event)
+{
+    if (event->type == SDL_WINDOWEVENT) {
+        event->type = PG_TO_WINDOWEVENT(*event);
+    }
+    return 1;
+}
+
 static void
 _pg_event_pump(int dopump)
 {
     if (dopump) {
         SDL_PumpEvents();
     }
-    /* We need to translate WINDOWEVENTS. But if we do that from the
-     * from event filter, internal SDL stuff that rely on WINDOWEVENT
-     * might break. So after every event pump, we translate events from
-     * here */
     SDL_FilterEvents(_pg_translate_windowevent, NULL);
```

Comment:
why isn't the filtering done in get then?

Reply:
...and in the `pg_event_wait` method probably too?

Reply:
This filter `_pg_translate_windowevent` called from `_pg_event_pump` runs in `clear`, `get` and `peek`; the kind of functions that deal with the entire event queue and not one at a time. Functions like `wait` and `poll` return atmost one event and don't filter for a particular type of event, so that is handled directly in `pgEvent_New`